File: gestalt/src/taskpane/services/claudeService.js
const API_URL = 'http://localhost:5000/api/chat';

export const sendMessageToClaude = async (message) => {
  if (!message.trim()) {
    throw new Error('Message cannot be empty');
  }

  try {
    console.log('Sending message to Claude:', message);
    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ message })
    });

    if (!response.ok) {
      const errorData = await response.json();
      throw new Error(errorData.error || 'Failed to get response from Claude');
    }

    const data = await response.json();
    return data.response;
  } catch (error) {
    console.error('Error calling Claude API:', error);
    throw error;
  }
}; 

File: gestalt/src/taskpane/services/claudeService.py
from anthropic import Anthropic
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Initialize the Anthropic client
client = Anthropic(
    api_key=os.getenv('ANTHROPIC_API_KEY', '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')
)

def send_message_to_claude(message):
    """
    Send a message to Claude and get the response.
    
    Args:
        message (str): The message to send to Claude
        
    Returns:
        str: Claude's response
    """
    try:
        # Create a message using the Claude API
        response = client.messages.create(
            model="claude-3-5-sonnet-latest",
            max_tokens=1000,
            messages=[
                {
                    "role": "user",
                    "content": message
                }
            ],
            temperature=0.7
        )
        
        # Extract the response text
        return response.content[0].text
        
    except Exception as e:
        print(f"Error calling Claude API: {str(e)}")
        raise 